components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/ApplicationBean.java

```diff
@@ -1129,6 +1129,9 @@ public void update(HttpServletRequest request) {
         String isSasApp = request.getParameter("isSaasApp");
         serviceProvider.setSaasApp((isSasApp != null && "on".equals(isSasApp)) ? true : false);
 
+        String isDiscoverableApp = request.getParameter("isDiscoverableApp");
+        serviceProvider.setDiscoverable(StringUtils.isNotBlank(isDiscoverableApp) && "on".equals(isDiscoverableApp));
```

Comment:
"on".equals(isDiscoverableApp) is enough.

Reply:
Fixed